### PR TITLE
Mlh0079 5284 add edl authentication

### PR DESCRIPTION
**Pull request by Michaellh0079**

### CODE REVIEW

File: cumulus_api/main.py [hunk 1]
```diff
@@ -1,50 +1,63 @@
...
-import requests
+from requests import Session
 
 from .cumulus_token import CumulusToken
 
 
 # pylint: disable=too-many-public-methods
 class CumulusApi:
-    def __init__(self, use_os_env: bool = True, config_path: str = None, token: str = None):
+    def __init__(self, token=None, config_file=None):
```
Comment (camposeddie) [inline cumulus_api/main.py]: `config_file` should remain `config_path` 
Updating will require py-cumulus-api users to update this variable name if already set within class instantiation and the update doesn't provide any additional context/functionality beyond a name change

**Comment by camposeddie** [comment]
This PR needs accompanying updates to `env.sh.example` and `config_sample.cfg`

**Review by camposeddie — CHANGES_REQUESTED** [review]
Missing env file updates

File: README.md [hunk 2]
```diff
@@ -27,27 +27,43 @@ from cumulus_api import CumulusApi
...
 Example of a config file passed to cumulus-api instance
 ```angular2html
 [DEFAULT]
-INVOKE_BASE_URL=************                     // Cumulus archive URL
-S3URI_LAUNCHPAD_CERT=****                       // s3 URI  to LAUNCHPAD pfx file
-LAUNCHPAD_PASSPHRASE_SECRET_NAME=*****         // LAUNCHPADPassphrase  secret id
-LAUNCHPAD_URL=*****                           // most likely https://api.launchpad.nasa.gov/icam/api/sm/v1/gettoken
-AWS_PROFILE=***                              // (Optional) Only if you want to run it locally
+INVOKE_BASE_URL=************       // Cumulus archive URL
+BASE_URL=************              // URS URL (https://uat.urs.earthdata.nasa.gov/)
```
Comment (camposeddie) [inline README.md]: defined here but not in env.sh.example

File: README.md [hunk 3]
```diff
@@ -27,27 +27,43 @@ from cumulus_api import CumulusApi
...
+
+# EDL
+CLIENT_ID=************************ // URS application id
+USER_NAME=************             // URS username
```
Comment (camposeddie) [inline README.md]: wrong variable name

File: README.md [hunk 4]
```diff
@@ -27,27 +27,43 @@ from cumulus_api import CumulusApi
...
+USER_PASSWORD=************         // URS password
```
Comment (camposeddie) [inline README.md]: wrong variable name

File: README.md [hunk 5]
```diff
@@ -27,27 +27,43 @@ from cumulus_api import CumulusApi
...
+
+# Launchpad
+LAUNCHPAD_CERT=****                // path to LAUNCHPAD pfx file
+LAUNCHPAD_PASSPHRASE=*****         // LAUNCHPAD Passphrase
+LAUNCHPAD_URL=*****                // most likely https://api.launchpad.nasa.gov/icam/api/sm/v1/gettoken
```
Comment (camposeddie) [inline README.md]: missing local certificate/passphrase variables

File: README.md [hunk 6]
```diff
@@ -27,27 +27,43 @@ from cumulus_api import CumulusApi
...
 Example of setting up environment variables
 ```code
-#!/usr/bash
-export INVOKE_BASE_URL=************                      // Cumulus archive URL
-export S3URI_LAUNCHPAD_CERT=****                        // s3 URI  to LAUNCHPAD pfx file
-export LAUNCHPAD_PASSPHRASE_SECRET_NAME=*****          // LAUNCHPADPassphrase  secret id
-export LAUNCHPAD_URL=*****                            // most likely https://api.launchpad.nasa.gov/icam/api/sm/v1/gettoken
-export AWS_PROFILE=***                               // (Optional) Only if you want to run it locally
+INVOKE_BASE_URL=************       // Cumulus archive URL
+BASE_URL=************              // URS URL (https://uat.urs.earthdata.nasa.gov/)
```
Comment (camposeddie) [inline README.md]: included here but not in actual env.sh.example file

File: README.md [hunk 7]
```diff
@@ -27,27 +27,43 @@ from cumulus_api import CumulusApi
...
+
+# EDL
+CLIENT_ID=************************ // URS application id
+USER_NAME=************             // URS username
```
Comment (camposeddie) [inline README.md]: see above comments for wrong variable names

File: README.md [hunk 8]
```diff
@@ -27,27 +27,43 @@ from cumulus_api import CumulusApi
...
+USER_PASSWORD=************         // URS password
+
+# Launchpad
+LAUNCHPAD_CERT=****                // path to LAUNCHPAD pfx file
+LAUNCHPAD_PASSPHRASE=*****         // LAUNCHPAD Passphrase
+LAUNCHPAD_URL=*****                // most likely https://api.launchpad.nasa.gov/icam/api/sm/v1/gettoken
 ```
-If you are running this tool locally you need to add your AWS_PROFILE name to the config
+If you are running this tool locally then you need to add your AWS_PROFILE name to the config.
```
Comment (camposeddie) [inline README.md]: not necessary for anything other than launchpad hosted credentials

File: config_sample.cfg [hunk 9]
```diff
@@ -1,10 +1,13 @@
...
+
+# EDL
 CLIENT_ID=************************ // URS application id
 USER_NAME=************             // URS username
 USER_PASSWORD=************         // URS password
-USE_LAUNCHPAD=*****                // set to "true" if using LAUNCHPAD anything else will use URS
+
+# Launchpad
 LAUNCHPAD_CERT=****                // path to LAUNCHPAD pfx file
 LAUNCHPAD_PASSPHRASE=*****         // LAUNCHPAD Passphrase
```
Comment (camposeddie) [inline config_sample.cfg]: wrong variable name and no inclusion of cloud hosted launchpad credentials

File: config_sample.cfg [hunk 10]
```diff
@@ -1,10 +1,13 @@
 [DEFAULT]
 INVOKE_BASE_URL=************       // Cumulus archive URL
 BASE_URL=************              // URS URL (https://uat.urs.earthdata.nasa.gov/)
+
+# EDL
 CLIENT_ID=************************ // URS application id
 USER_NAME=************             // URS username
```
Comment (camposeddie) [inline config_sample.cfg]: see above wrong variable name issue

File: env.sh.example [hunk 11]
```diff
@@ -1,6 +1,13 @@
 #!/usr/bash
 export AWS_PROFILE=[REDACTED]
 export INVOKE_BASE_URL=[REDACTED]
+
+# EDL
+export CLIENT_ID=[REDACTED]
+export USER_NAME=[REDACTED]
+export USER_PASSWORD=[REDACTED]
+
+# Launchpad
```
Comment (camposeddie) [inline env.sh.example:14]: missing local launchpad variables

File: env.sh.example [hunk 12]
```diff
@@ -1,6 +1,13 @@
 #!/usr/bash
 export AWS_PROFILE=[REDACTED]
 export INVOKE_BASE_URL=[REDACTED]
+
+# EDL
+export CLIENT_ID=[REDACTED]
+export USER_NAME=[REDACTED]
```
Comment (camposeddie) [inline env.sh.example]: wrong variable names

File: README.md (hunk 8 above)
Comment (camposeddie) [reply README.md]: Is `AWS_PROFILE` needed locally if you are using EDL credentials or locally hosted launchpad credentials?

Comment (Michaellh0079) [reply README.md]: I don't believe so. I did not add this statement but only corrected the grammar. If you could capture any other additional requests you have into a single list and post them I will address them.

Comment (camposeddie) [reply README.md]: Everything else looks good 👍

**Review by camposeddie — APPROVED** [review]

**Pull request merged**